googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -380,14 +389,56 @@ class GTEST_API_ Mock {
   // Verifies and clears all expectations on the given mock object.
   // If the expectations aren't satisfied, generates one or more
   // Google Test non-fatal failures and returns false.
-  static bool VerifyAndClearExpectations(void* mock_obj)
-      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex);
+  template <class M>
+  static bool VerifyAndClearExpectations(M* mock_obj)
+      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+    return VerifyAndClearExpectationsImpl(mock_obj);
+  }
+
+  template <class M>
+  static bool VerifyAndClearExpectations(NiceMock<M>* mock_obj)
+      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+    return VerifyAndClearExpectationsImpl(internal::ImplicitCast_<M*>(mock_obj));
+  }
+
+  template <class M>
+  static bool VerifyAndClearExpectations(NaggyMock<M>* mock_obj)
+      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+    return VerifyAndClearExpectationsImpl(internal::ImplicitCast_<M*>(mock_obj));
+  }
+
+  template <class M>
+  static bool VerifyAndClearExpectations(StrictMock<M>* mock_obj)
+      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex) {
+    return VerifyAndClearExpectationsImpl(internal::ImplicitCast_<M*>(mock_obj));
+  }
 
   // Verifies all expectations on the given mock object and clears its
   // default actions and expectations.  Returns true iff the
   // verification was successful.
-  static bool VerifyAndClear(void* mock_obj)
-      GTEST_LOCK_EXCLUDED_(internal::g_gmock_mutex);
+  template <class M>
+  static bool VerifyAndClear(M* mock_obj)
```

Comment:
Just constrain VerifyAndClear to only accept a Matcher*.
I'd try an `enable_if` for that.